lib/view/pages/events/view_events_page.dart

```diff
@@ -82,373 +94,173 @@ class _HomePageWidgetState extends State<ViewEvents> {
                         return Center(
                           child: LoadingAnimationWidget.halfTriangleDot(
                             color: Theme.of(context).colorScheme.primary,
-                            size: 36.pixelScale(context),
+                            size: 36,
                           ),
                         );
                       } else if (snapshot.hasError) {
                         return Text('Error: ${snapshot.error}');
                       } else if (snapshot.hasData) {
-                        allEvents = [];
-                        for (var x in snapshot.data!.docs) {
-                          var data = x.data() as Map<String, dynamic>;
-                          Event event = Event.fromMap(data, x.id);
-                          allEvents?.add(event);
+                        if (allEvents == null) {
```

Comment:
I'm sorry I have to point this out again. Could you please leave in my code here ? 
This is what my code does:
Stream gets data everytime something changes on the database. Then I empty allEvents , so that allEvents always has the most up to date data from the stream. Then I put in the newest data from the stream into the allEvents object. I don't feel a check if allevents == null does the same. The way you did it means that the data stys the same until you refresh. My way always got the most up to date data when something changed, yes there are some weird things that don't immediatelly update ( I don't know why ) but a lot of things do update. Please put my code back in here.

Reply:
Fixed